src/main/java/seedu/taskman/logic/commands/CompleteCommand.java

```diff
@@ -39,53 +40,62 @@ public CompleteCommand(int targetIndex) {
 	public CommandResult execute() {
 		assert model != null;
 		
-		UnmodifiableObservableList<Activity> lastShownList = model.getFilteredActivityList();
-
-        if (lastShownList.size() < targetIndex) {
-            indicateAttemptToExecuteIncorrectCommand();
-            return new CommandResult(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
-        }
-
-        Activity activityToComplete = lastShownList.get(targetIndex - 1);
-        activityType = activityToComplete.getType();
-        
-        switch (activityType){
-	        case TASK:
-	        default: {
-	        	Task task = new Task(
-	                    activityToComplete.getTitle(),
-	                    activityToComplete.getTags(),
-	                    activityToComplete.getDeadline().get(),
-	                    activityToComplete.getSchedule().get(),
-	                    activityToComplete.getFrequency().get());
-	        	try {
-					task.setStatus(new Status(STATUS_COMPLETE));
-				} catch (IllegalValueException e) {
-					// TODO Auto-generated catch block
-					e.printStackTrace();
-				}
-	            afterComplete = new Activity(task);
-	        }
+		try {
+            initMembers();
+        } catch (IllegalValueException e) {
+            return new CommandResult(e.getMessage());
         }
         		
         try {
             model.deleteActivity(activityToComplete);
             model.addActivity(afterComplete);
             return new CommandResult(String.format(MESSAGE_SUCCESS, afterComplete.getTitle().title));
         } catch (UniqueActivityList.ActivityNotFoundException pnfe) {
-
             indicateAttemptToExecuteIncorrectCommand();
             return new CommandResult(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
-
         } catch (UniqueActivityList.DuplicateActivityException e) {
-
             try {
```

Comment:
Not part of your code, but do we really need to try to add the activity again, when the adding has failed previously (and resulted in the current exception?)

Seems like we can make do without this nested try block.